core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/BaseJournalDaoWithReadMessages.scala

```diff
@@ -51,17 +51,24 @@ trait BaseJournalDaoWithReadMessages extends JournalDaoWithReadMessages {
       toSequenceNr: Long,
       batchSize: Int,
       refreshInterval: Option[(FiniteDuration, Scheduler)]) = {
+    val firstSequenceVer: Long = Math.max(1, fromSequenceNr)
     Source
-      .unfoldAsync[(Long, FlowControl), Seq[Try[(PersistentRepr, Long)]]]((Math.max(1, fromSequenceNr), Continue)) {
+      .unfoldAsync[(Long, FlowControl), Seq[Try[(PersistentRepr, Long)]]]((firstSequenceVer, Continue)) {
         case (from, control) =>
-          def limitWindow(from: Long): Long = {
-            math.min(from + batchSize, toSequenceNr)
+          def limitWindow(from: Long)(implicit fallback: Boolean): Long = {
```

Comment:
I am not sure if this parameter should be implicit or not, And I will adopt the appropriate suggestions and modify it.